irg_gazebo_plugins/materials/scripts/IRGCameraSim.frag

```diff
@@ -157,9 +162,13 @@ void main()
   // convert light power to sensor signal
   color.rgb *= vec3(energy_conversion);
 
+  // Number of possible values in the final image encoded as integers, or the
+  // analog-to-digital limit.
+  float integer_limit = pow(2.0, adc_bits);
```

Comment:
Should this instead be? Or is it intentional?
```suggestion
  float integer_limit = pow(2.0, adc_bits) - 1;
```

Reply:
This is intentional, though Larry and I have debated this quite a bit. This value should be the number of discrete steps in the final image.

Reply:
It does seem that you or whoever was the original author of this fragment was actually using pow(2.0, adc_bits) - 1 within the **gaussrand** method. Since the value you were multiplying/dividing with was 4095.0. But with the current implementation and for adc_bits=12 you are using 4096 instead. This will slightly change the final outcome.. very slightly.

Reply:
Very slightly :)